algebra-core/src/fields/mod.rs

```diff
@@ -158,10 +160,30 @@ pub trait Field:
     }
 }
 
-/// A trait that defines parameters for a prime field.
-pub trait FpParameters: 'static + Send + Sync + Sized {
+/// A trait that defines parameters for a field that can be used for FFT.
+pub trait FftParameters: 'static + Send + Sync + Sized {
     type BigInt: BigInteger;
 
+    /// 2^s * t = MODULUS - 1 with t odd. This is the two-adicity of
+    /// `Self::MODULUS`.
+    const TWO_ADICITY: u32;
+
+    /// 2^s root of unity computed by GENERATOR^t
+    const ROOT_OF_UNITY: Self::BigInt;
```

Comment:
Maybe for clarity we should rename this to `TWO_ADIC_ROOT_OF_UNITY` or `TWO_ADIC_SUBGROUP_GENERATOR`? Currently it's not clear from the name what the difference is between `ROOT_OF_UNITY` and `FULL_ROOT_OF_UNITY`

Reply:
Yes, I like that idea!